oop2-encapsulate-inherit.js

```diff
@@ -5,27 +5,269 @@
+        getAuthor(){
+            return this.#author;
+        }
+
+        getPublisher(){
+            return this.#publisher;
+        }
+
+        setAuthor(author){
+            this.#author = author;
+        }
+
+        setPublisher(publisher){
+            this.#publisher = publisher;
+        }
+    }
+
+    class Author {
+        #name;
+        #books;
+
+        constructor(name, books){
+            this.#name = name;
+            this.#books = books;
+        }
+
+        getName(){
+            return this.#name;
+        }
+
+        getBooks(){
+            return this.#books;
+        }
+
+        setName(name){
+            this.#name = name;
+        }
+
+        setBooks(books){
+            this.#books = books;
+        }
+    }
+
+    class Publisher {
+        #authors;
+        #books;
 
+        constructor(authors, books){
+            this.#authors = authors;
+            this.#books = books;
+        }
 
+        getAuthors(){
+            return this.#authors;
+        }
 
+        getBooks(){
+            return this.#books;
+        }
 
+        setAuthors(authors){
+            this.#authors = authors;
+        }
 
+        setBooks(books){
+            this.#books = books;
+        }
+    }
 
-2. Create the following classes:
+    class Review {
+        #rating;
+        #user;
+
+        constructor(rating, user){
+            this.#rating = rating;
+            this.#user = user;
+        }
+
+        getRating(){
+            return this.#rating;
+        }
+
+        getUser(){
+            return this.#user;
+        }
+
+        setRating(rating){
+            this.#rating = rating;
+        }
+
+        setUser(user){
+            this.#user = user;
+        }
+    }
+
+
+
+
+
+
+/* 2. Create the following classes:
     - Umbrella class.  It should have a an organization name.  It should only have a single instance.  Umbrella is the "Parent Organization" e.g. Allegis Group
     - Company class. It should have a compnay name and # of employees.  Create at least 3 different companies that are children of the Umbrella Organization.  (e.g. TEKsystems, Aerotek, etc.)
     - Site class. It should have the name of the company, and the location of the site.  It is a child class that inherits from the Company class.  Create 3 sites for each company.
     - Employee class. It should have a employee name, job title, and salary properties. Create 10 employee instances: 1 CEO, 1 Manager, 1 Secretary, 2 Engineers, 1 Financial Officer, 1 Janitor, 1 Marketer, 1 HR Personnel, 1 Lawyer.  Feel free to add other employee roles at your own discretion.
     
     - For each class, add 2 properties and 2 methods to each.  The properties and methods should make sense, considering what the classes are supposed to be representing.
-
+*/
 
 
     //your code here...
 
+    class Umbrella {
+        #orgName;
+        #yearFormed = 1999;
+        #listOfCompanies = ['TEKsystems', 'Aerotek', 'TEK Global'];
+        #updatedList;
```

Comment:
Why initialize the fields for `yearFormed` and `listOfCompanies`, but not `orgName`? Are you suggesting you can add any organization name here and they'd have been founded in 1999 if you don't otherwise explicitly state? Seems off.